addon/components/g-map-infowindow.js

```diff
@@ -43,10 +44,30 @@ const GMapInfowindowComponent = Ember.Component.extend({
     }
   },
 
+  optionsChanged: observer('disableAutoPan', 'maxWidth', 'pixelOffset', function() {
+    run.once(this, 'setOptions');
+  }),
+
+  setOptions() {
+    const infowindow = this.get('infowindow');
+    const options = ['disableAutoPan', 'maxWidth', 'pixelOffset'];
```

Comment:
It could be, but there is a setZindex method and i thought that it should be implemented separately for consistency, as you already stated above. It could be added when the rest of the options are refactored too. I wasn't sure if all the options should be set this way to be honest. Your call